### DISCUSSION TITLE
A config inspector

### DISCUSSION THREAD

**Discussion by Airkro** [renovatebot/renovate#11086] [Suggest an Idea]
When using `packageRules`, that always something wrong, sometimes `semanticCommitType` does not work as expected, sometimes `minor: { enable: false }` does not work. 

Config merging logic is a totally black box, although we can use `printConfig: true` to reproduce, run `renovate` locally is too difficult, I guess 90%+ people will quit.

We need an inspector to see what happens, what our configuration really became.

**Comment by rarkins** [comment]
I agree that this would be nice. Being written in JS/TS, I think we can make this happen somehow in future in the browser. There's two key caveats though:
1. Our config language is very powerful/expressive, so it's a little like SQL in that there are many ways to get it wrong and a config inspector won't be able to magically guess/detect them all
2. I think what you'd benefit more from is not just a config inspector but a contextual one. e.g. something which takes not just your config but also a copy of your last extracted dependencies so that you can see the effect on your repo by changes

**Reply by Airkro** [reply]
> I think what you'd benefit more from is not just a config inspector but a contextual one. e.g. something which takes not just your config but also a copy of your last extracted dependencies so that you can see the effect on your repo by changes

You mean a playground, that sounds nice.

**Reply by HonkingGoose** [reply]
Related issues:

- https://github.com/renovatebot/renovate/issues/547
- https://github.com/renovatebot/renovate/issues/1502
